File: src/main/java/com/mamaliang/npps/cas/server/ticket/ExpirationPolicy.java
package com.mamaliang.npps.cas.server.ticket;

/**
 * ticket过期策略接口
 */
public interface ExpirationPolicy {

  /**
   * 获取ticket允许的最大存活时间
   *
   * @return 最大存活时间，单位秒
   */
  Long getTimeToLive();

  /**
   * 获取ticket允许的最大空闲时间
   *
   * @return 最大空闲时间，单位秒
   */
  Long getTimeToIdle();

  /**
   * 获取过期策略名称
   *
   * @return 名称
   */
  String getName();
}

File: src/main/kotlin/com/mamaliang/npps/cas/server/ticket/ExpirationPolicy.kt
package com.mamaliang.npps.cas.server.ticket

/**
 * ticket过期策略接口
 */
interface ExpirationPolicy {
    /**
     * ticket允许的最大存活时间,单位秒
     */
    val timeToLive: Long

    /**
     * ticket允许的最大空闲时间
     *
     * @return 最大空闲时间，单位秒
     */
    val timeToIdle: Long

    /**
     * 过期策略名称
     */
    val name: String
}
